cleo_plugins/Audio/Audio.cpp

```diff
@@ -0,0 +1,331 @@
+        auto handle = OPCODE_READ_PARAM_PED_HANDLE();
+
+        auto ped = CPools::GetPed(handle);
+        stream->Link(ped);
+
+        return OR_CONTINUE;
+    }
+
+    //0AC5=2,link_3d_audiostream %1d% to_vehicle %2d%
+    static OpcodeResult __stdcall opcode_0AC5(CScriptThread* thread)
+    {
+        auto stream = (CAudioStream*)OPCODE_READ_PARAM_PTR(); VALIDATE_STREAM();
+        auto handle = OPCODE_READ_PARAM_VEHICLE_HANDLE();
+
+        auto vehicle = CPools::GetVehicle(handle);
+        stream->Link(vehicle);
+
+        return OR_CONTINUE;
+    }
+
+    //2500=1,  is_audio_stream_playing %1d%
+    static OpcodeResult __stdcall opcode_2500(CScriptThread* thread)
+    {
+        auto stream = (CAudioStream*)OPCODE_READ_PARAM_PTR(); VALIDATE_STREAM();
+
+        auto state = stream->GetState();
+
+        OPCODE_CONDITION_RESULT(state == 1);
+        return OR_CONTINUE;
+    }
+
+    //2501=2,%2d% = get_audiostream_length_fine %1d%
+    static OpcodeResult __stdcall opcode_2501(CScriptThread* thread)
+    {
+        auto stream = (CAudioStream*)OPCODE_READ_PARAM_PTR(); VALIDATE_STREAM();
+
+        auto length = stream->GetLength();
+        length /= max(stream->GetSpeed(), 0.000001f); // speed corrected
+
+        OPCODE_WRITE_PARAM_FLOAT(length);
+        return OR_CONTINUE;
+    }
+
+    //2502=2,get_audio_stream_speed %1d% store_to %2d%
+    static OpcodeResult __stdcall opcode_2502(CScriptThread* thread)
+    {
+        auto stream = (CAudioStream*)OPCODE_READ_PARAM_PTR(); VALIDATE_STREAM();
+
+        auto speed = stream->GetSpeed();
+
+        OPCODE_WRITE_PARAM_FLOAT(speed);
+        return OR_CONTINUE;
+    }
+
+    //2503=2,set_audio_stream_speed %1d% speed %2d%
+    static OpcodeResult __stdcall opcode_2503(CScriptThread* thread)
+    {
+        auto stream = (CAudioStream*)OPCODE_READ_PARAM_PTR(); VALIDATE_STREAM();
+        auto speed = OPCODE_READ_PARAM_FLOAT();
+
+        stream->SetSpeed(speed);
+
+        return OR_CONTINUE;
+    }
+
+    //2504=3,set_audio_stream_volume_with_transition %1d% volume %2d% time_ms %2d%
+    static OpcodeResult __stdcall opcode_2504(CScriptThread* thread)
+    {
+        auto stream = (CAudioStream*)OPCODE_READ_PARAM_PTR(); VALIDATE_STREAM();
+        auto volume = OPCODE_READ_PARAM_FLOAT();
+        auto time = OPCODE_READ_PARAM_INT();
+
+        stream->SetVolume(volume, 0.001f * time);
+
+        return OR_CONTINUE;
+    }
+
+    //2505=3,set_audio_stream_speed_with_transition %1d% speed %2d% time_ms %2d%
+    static OpcodeResult __stdcall opcode_2505(CScriptThread* thread)
+    {
+        auto stream = (CAudioStream*)OPCODE_READ_PARAM_PTR(); VALIDATE_STREAM();
+        auto speed = OPCODE_READ_PARAM_FLOAT();
+        auto time = OPCODE_READ_PARAM_INT();
+
+        stream->SetSpeed(speed, 0.001f * time);
+
+        return OR_CONTINUE;
+    }
+
+    //2506=2,set_audio_stream_source_size %1d% radius %2d%
+    static OpcodeResult __stdcall opcode_2506(CScriptThread* thread)
```

Comment:
since this is a property of 3D stream the opcode should be 
```
set_3d_audio_stream_source_size
```
and belong to class [AudioStream3D](https://library.sannybuilder.com/#/sa/classes/AudioStream3D)

Reply:
also consider alternative name: `set_3d_audio_stream_distance`

Reply:
This is used to set 3d sound source sphere radius.  "distance" would suggest this is related to max hearing distance.